packages/synthetic-chain/README.md

```diff
@@ -18,12 +10,47 @@ node_modules/.bin/synthetic-chain test
 node_modules/.bin/synthetic-chain test --debug -m <substring of proposal name>
 ```
 
-shared JS is exported from the package.
-non-JS is in `files` and can be untarred out for use in a3p
+## Design
+
+It builds images starting from ag0 or an already build synthetic-chain. The build stages sequence is produced by `dockerfileGen.ts` approximately as so:
+
+```mermaid
+---
+title: Build Stages
+---
+flowchart TD
+    M{Mode?}
+    M -- FromAg0 --> S[START]
+    M -- Append --> R[RESUME]
+    S --> NP{Proposal type}
+    R --> NP
+    NP -- Software Upgrade Proposal ----> Pr[PREPARE]
+    Pr ----> Exec[EXECUTE]
+    Exec ----> Use[USE]
+    Use ----> Test[TEST]
+    NP -- CoreEvalProposal ----> Eval[EVAL]
+    Eval ----> Use
+    Test ----> AP{Another proposal?}
+    AP -- Yes ----> NP
+    AP -- Yes ----> END[DEFAULT last use]
+```
+
+## Development
 
 To depend on `@agoric/synthetic-chain` that isn't yet published, use `npm pack` in this package and copy the tgz into the proposal. Then use the `file:` protocol in the package.json to add it. Finally `yarn install` in the package to update local node_modules for linting. E.g.,
 
 ```json
     "dependencies": {
         "@agoric/synthetic-chain": "file:agoric-synthetic-chain-0.0.1-alpha.tgz",
 ```
+
+### Debugging proposals
+
+The build will fail if an upgrade proposal fails or but a CoreEval. That's
```

Comment:
Two things.
 * I suspect this triple-quote is unclosed
 * something's wrong with the wording at the end of the sentence.

Reply:
> suspect this triple-quote is unclosed

I see three opens and three closes.

> something's wrong with the wording at the end of the sentence

thx fixed

Reply:
It looks like github is unhappy with the diff. I dunno why.

![Screenshot 2024-01-12 at 10 02 23 AM](https://github.com/Agoric/agoric-3-proposals/assets/13372636/56ba3dc8-72ab-4963-b061-c58ae3201e8f)

Reply:
oh the differ is confused. you can see the full file renders correctly https://github.com/Agoric/agoric-3-proposals/blob/8adbcf63ed9dda0727c13ee8fe4d135597663ccb/packages/synthetic-chain/README.md